### ISSUE TITLE
Improve robustness, markup and parsing of build log on Travis

### ISSUE THREAD

**Issue by heplesser** [nest/nest-simulator#392]
The work on #380 and #391 revealed that the build log generated on Travis should be improved:
- Instead of relying on text or commands, introduce explicit marker lines with special format for mark the beginning and end of certain sections, e.g. `======= BEGIN VERA =======` and  `======= END VERA =======`. This will allow for much more robust parsing, even as the content of sections evolves.
- If the statistical analysis has found problems, the final report may contain the entire diff result. Thus, the summary becomes very difficult to read. I suggest to keep it shorter, at least not include the full diff if it is longer than a few lines.
- On rare occasions, e.g. [Travis Build 841.4](https://travis-ci.org/nest/nest-simulator/jobs/136142790) the 

```
curl https://api.github.com/repos/nest/nest-simulator/pulls/389/files
```

  in [build.sh](https://github.com/nest/nest-simulator/blob/ef980a92fef66d9d5a43429eb987f851c3a607a6/build.sh#L117) may fail, presumably if Github cannot fullfill the request on time, leading to a 

```
jq: error (at <stdin>:4): Cannot index string with string "filename"
```

In that case, the `file_names` variable in `build.sh` remains empty, no files are tested for formatting errors, and the build may pass even if the PR contained files with formatting errors. This should not happen: if the command filling `file_names` fails, the build should stop.

**Comment by tammoippen** [comment]
Please see #395 if it fixes most of the points.

**Comment by heplesser** [comment]
@gtrensch Given your suggestions for further improvements, I leave this issue open even after #395 is merged and leave it in your care.

**Comment by gtrensch** [comment]
This issue has been resolved by #546 and can be closed.
